src/main/java/org/apache/solr/benchmarks/BenchmarksMain.java

```diff
@@ -105,32 +108,124 @@ public static void runQueryBenchmarks(List<QueryBenchmark> queryBenchmarks, Stri
 		    for (int threads = benchmark.minThreads; threads <= benchmark.maxThreads; threads++) {
 		        QueryGenerator queryGenerator = new QueryGenerator(benchmark);
 		        HttpSolrClient client = new HttpSolrClient.Builder(baseUrl).build();
+		        String collection = collectionNameOverride==null? benchmark.collection: collectionNameOverride;
 		        ControlledExecutor controlledExecutor = new ControlledExecutor(threads,
 		                benchmark.duration,
 		                benchmark.rpm,
 		                benchmark.totalCount,
 		                benchmark.warmCount,
-		                getQuerySupplier(queryGenerator, client, collectionNameOverride==null? benchmark.collection: collectionNameOverride));
+		                getQuerySupplier(queryGenerator, client, collection));
 		        long start = System.currentTimeMillis();
+						List<SolrBenchQuery> generatedQueries;
 		        try {
-		            controlledExecutor.run();
+		            generatedQueries = controlledExecutor.run();
 		        } finally {
 		            client.close();
 		        }
 
+	        	int success = 0, failure = 0;
+
+		        if (StressMain.generateValidations) {
+		        	int totalDocsIndexed = getTotalDocsIndexed(baseUrl, collection);
+		        			
+		        	FileWriter outFile = new FileWriter("suites/validations-"+benchmark.queryFile+"-docs-"+totalDocsIndexed+"-queries-"+benchmark.totalCount+".tsv");
+			        for (SolrBenchQuery sbq: generatedQueries) {
```

Comment:
This might not work properly, I assume the expectation is that we want every single executed queries during the test and verify them all? If so, then it might not work as instead of return a new instance of `QueryRequest` for each call as in https://github.com/fullstorydev/solr-bench/pull/44/files#diff-e8aa24ed8e0a14385ea7fdfa11268a85f6dcce52021610761f29a76c334e558bL57, it's now returning a saved instance of the stored field in `QueryGenerator` https://github.com/fullstorydev/solr-bench/pull/44/files#diff-e8aa24ed8e0a14385ea7fdfa11268a85f6dcce52021610761f29a76c334e558bR73 . Therefore calls that uses the same query will overwrite previous results.

However, I do have some concerns if we correct this by generating new `SolrBenchQuery` instance per call, as the design right now store both the underlying `QueryRequest` and the string response and those are kept until the very end to generate the validation file, which before such change both `QueryRequest` and the response string has much shorter lifespan (just for the local call and can get GCed). So if we have a huge number of query runs, we could run into memory problem with the new design.